### PR TITLE
Use null coalescing operator to fix notice

### LINKED ISSUE #25782: BO - Products page - an exception is displayed when combination feature is disabled
#### Describe the bug

After the update from 1.7.7.5 to 1.7.7.7 the error 'Undefined index: attribute_price' appears.
#### Steps to Reproduce

1. Enable debug mode
2. Delete all products with combinations
3. Go to BO > Advanced Parameters > Performance page
4. Disable Combination
5. Go to BO > Catalog > Products page
6. See an error  
image

It is ok with PS1775

---------------------------------------------------------------------------

It fails on line 3418 (1.7.7.7) in Product.php. 
`'attribute_price' => $row['attribute_price'] ?: null, `

Bevore line 3408 (1.7.7.5) there was this code that works:
`'attribute_price' => (isset($row['attribute_price']) ? $row['attribute_price'] : null),`

**It fails only if the feature Combination is disabled.**

I don't understand because this code works in the same environment:
```
$array2 = [
    "foo" => "bar",
    "bar" => "foo",
];

echo $array2['foos'] ?: "gugus";
```

* PrestaShop version: 1.7.7.7
* PHP version: 7.2.24 on Ubuntu

### PR DESCRIPTION
**Pull request by kpodemski**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 1.7.7.x
| Description?      | fix undefined var and possible notice
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #25782
| How to test?      | Enable debug mode, disable combinations
| Possible impacts? | Please indicate what parts of the software we need to check to make sure everything is alright.
| Sponsor company | impSolutions

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25784)

### CODE REVIEW

**Comment by kpodemski** [comment]
CI is failing because of .com being not available.

**Review by atomiix — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thank you @kpodemski

**Comment by Progi1984** [comment]
@kpodemski Sorry. But before sending it in QA, we need to know if @PrestaShop/product-team target the issue for 177x or 178x.

**Comment by MatShir** [comment]
It is for the 1.7.7.8 release ! Go go

**Comment by Robin-Fischer-PS** [comment]
Hi,

I'm having an issue with the PR, following these steps : 

1. After install, change all combinations products to simple products : OK
2. In Advanced Parameters > Performance : Disable combination products : OK
3. In International > Taxes : Enable Ecotax : OK
4. In Advanced Parameters > Performance : Enable combination products : OK
5. Edit a simple product to change it to combination product (with at least two combinations) : OK
6. In each combination : Add an Ecotax, save : OK
7. Change this combination product to simple product again : OK
8. In Advanced Parameters > Performance : Disable combination products in unavailable : KO
9. Delete the product : Still impossible to disable combination products in Advanced Parameters : KO

I would like to have a @PrestaShop/qa-functional to reproduce this error : I cannot reproduce it without the PR.

**Comment by matks** [comment]
Hello everyone,

I have explored this PR as much as I could this morning.
So I tried many things, and I have a theory to explain why the following behavior is sometimes reproduced and sometimes not:

When you open a "Edit a Product" BO page of a product that has combinations, you'll notice that upon loading, the page has a first look then a few seconds later it changes! to display a 2nd look. The 2nd look has the combinations tab.

This happens because the look of the Product page is controlled by JavaScript which requires a few seconds to load. So until the JavaScript has loaded and initialized fully, the page is in a weird state. 😮 

I think that sometimes, because you do this quickly, you start clicking on buttons and inputs on this page _before the page has fully loaded_, _before the JavaScript is fully loaded_ and this results in incorrect behavior. I think sometimes you open the page and immediately you click on "product simple" to convert a product with combinations into a standard product, and since you do this so fast, you actually trigger the "convert a product with combinations into a standard product" too soon. Then it does weird things 😨  .

I think we have at least 2 consequences of this weird behavior:
- the pop-up to confirm that the action will delete the combinations does not show
- the combinations are badly and partially deleted, the data becomes corrupted (and you can see a symptom of that in SQL table ps_product_attribute)

When products are incorrectly converted from product with combinations into a standard product, the "Advanced Parameters > Performance : Disable combination products" cannot be clicked.

This is because the "Advanced Parameters > Performance : Disable combination products" can be clicked when PrestaShop verifies there is no combinations in the shop. But in the above scenario, combinations have been partially deleted!

To summarize:
- when you open a "Edit a product" BO page you should wait for the page to fully load
- if you do not wait, and you immediately start using the page to convert a product with combinations into a standard product, the process of conversion might be corrupted
- when there is corrupted combination data, the "Advanced Parameters > Performance : Disable combination products" cannot be clicked

The PR is OK in my opinion, the issues that @Robin-Fischer-PS observed are related to the above "click too fast" scenario, not the PR.

**Comment by khouloudbelguith** [comment]
Hi @kpodemski,

I checked with @Robin-Fischer-PS.
This PR is ok :heavy_check_mark: 

As discussed with @matks https://github.com/PrestaShop/PrestaShop/issues/25808 is not blocking this PR.
Some tests:

1. Before disabling combination: create order with a product with combinations
2. Combination feature disabled, create all types of products with retail price/cost price/price unit/ecotax / specific price/supplier price
3. Create an order with those products / create an order with a product with combinations transformed to a simple product
4. Generate invoice/credit slip/delivery slip
5. Check with multistore (disable / enable combinations feature in all shops or in different shop)
6. Change the default currency / create an order with new currency
7. Import products

https://drive.google.com/file/d/1kfieI4_OpJe0S8ikl9oOC7bPP9kLYSri/view?usp=sharing

Thanks!

**Comment by matks** [comment]
Thank you @kpodemski

**Comment by matks** [comment]
Thank you @Robin-Fischer-PS and @khouloudbelguith for the careful validation

**Comment by kpodemski** [comment]
Thank you @matks for the investigation ❤️
